pkg/generic/httppbthrift_codec.go

```diff
@@ -143,18 +143,9 @@ func (c *httpPbThriftCodec) Close() error {
 // FromHTTPPbRequest parse  HTTPRequest from http.Request
 func FromHTTPPbRequest(req *http.Request) (*HTTPRequest, error) {
 	customReq := &HTTPRequest{
-		Header:      req.Header,
-		Query:       req.URL.Query(),
-		Cookies:     descriptor.Cookies{},
-		Method:      req.Method,
-		Host:        req.Host,
-		Path:        req.URL.Path,
+		Request:     req,
 		ContentType: descriptor.MIMEApplicationProtobuf,
 	}
-	for _, cookie := range req.Cookies() {
-		customReq.Cookies[cookie.Name] = cookie.Value
-	}
```

Comment:
Also the possible performace case for Cookies. Here all cookies are set to a map for faster read; the new implementation called `http.Request.Cookie(name)`, which will always parse all the cookies each time it's called.

Reply:
made cache for cookies.